map/framework.cpp

```diff
@@ -3313,3 +3317,81 @@ void Framework::OnPowerSchemeChanged(power_management::Scheme const actualScheme
   if (actualScheme == power_management::Scheme::EconomyMaximum && GetTrafficManager().IsEnabled())
     GetTrafficManager().SetEnabled(false);
 }
+
+bool Framework::ShouldShowProducts() const
+{
+  auto const connectionStatus = Platform::ConnectionStatus();
+  if (connectionStatus == Platform::EConnectionType::CONNECTION_NONE)
+    return false;
+
+  std::string donateUrl;
+  if (!settings::Get(settings::kDonateUrl, donateUrl)) // donation is disabled
+    return false;
+
+  if (!m_usageStats.IsLoyalUser())
+    return false;
+
+  if (!storage::IsPointCoveredByDownloadedMaps(GetCurrentPlacePageInfo().GetMercator(), m_storage, *m_infoGetter))
+    return false;
+
+  #ifdef DEBUG
+  uint32_t constexpr kPopupCloseTimeout = 10;
+  uint32_t constexpr kProductSelectTimeout = 20;
+  uint32_t constexpr kRemindMeLaterTimeout = 5;
+  #else
+  uint32_t constexpr kPopupCloseTimeout = 60 * 60 * 24  * 14; // 14 days
```

Comment:
The opinion poll in OM dev chats showed that weighted average preference is 2.4 months (`(0.5 * 3 + 1 * 3 + 3 * 4 + 6 * 2) / 12 = 2.38`) (based on votes of 12 developers).

I guess we can round it to 2.5 months / 75 days?

Reply:
> This particular constant in the code is for "X" (Close) button that plays "Remind me later" function.

Did you forget there is a separate "Remind Later" button which is set to 3 days (as per your own suggestion https://github.com/organicmaps/organicmaps/pull/9695#discussion_r1855484906) ? :)

The "X" (close) button is for other reasons, e.g. "Don't want to donate".

But please let's not repeat ourselves with the same arguments.

Developers have different opinions on this topic and we couldn't come to an agreement, that's why we had a poll.